### ISSUE TITLE
Regression: input multiple domains in domain input field adds them as a single domain

### ISSUE THREAD

**Issue by yubiuser** [pi-hole/AdminLTE#2674]
Expected behavior is like it was in v5: domains separated by space should be added as individual entries.

**Comment by rdwebdesign** [comment]
I'm not sure if this issue should be handle in the web interface or in FTL.  

In `v5`, the first step executed by PHP is to split the input into an array of domains:

https://github.com/pi-hole/AdminLTE/blob/41682f17b72d3fb83837a7a08fa68b3e37cd35b7/scripts/pi-hole/php/groups.php#L534C9-L534C77

In this case, the expected value for the domain field (`$_POST['domain']`) is a string with one or more domains separated by spaces.

The current `v6` web interface doesn't use a server side code here. It sends the whole text to the API. FTL doesn't seems to take care of spaces or line breaks and the whole string is used.

**Comment by yubiuser** [comment]
> FTL doesn't seems to take care of spaces or line breaks and the whole string is used.

Exactly, FTL expects one domain to be added per request. I think the web interface needs to iterate over the input data and split it. Can this be done in JS?

**Comment by rdwebdesign** [comment]
> Can this be done in JS?

Yes, but this will send multiple API requests (one for each domain) and FTL will reload the list after each request.

I was thinking if this should be done in FTL, to handle multi-domain inputs (from web interface or command line) in a single step and reload the lists only once.

**Comment by yubiuser** [comment]
> API requests (one for each domain) and FTL will reload the list after each request

I see the issue here. Agree, this should be handled by FTL
